### ISSUE TITLE
Add support for building go-libtor on Android, Linux and iOS

### ISSUE THREAD

**Issue by hellais** [ooni/probe#1867]
This involves generating the orconfig.h for Android, Linux and iOS so as to make the build work.

As part of this task I will also move my fork of go-libtor over to the OONI org: https://github.com/hellais/go-libtor/tree/update/tor-0.4.6.7

**Comment by hellais** [comment]
I have updated go-libtor to add support for building on android and ios. Following the discussion with @bassosimone we have decided that for linux we are going to support running tor by simply depending on the tor package and spawning the tor process so we don't actually need support for that.

In order to generate the needed orconfig.h files for android and ios I used the scripts in this repo: https://github.com/ooni/cross-build-tor. It's not 100% automated, but if we ever need to regenerate them it should be pretty quick and straightforward as most of the tricky parts of figuring out how to get the cross compilers to work is sorted.

I also went ahead and created a fork of go-libtor in the ooni namespace and updated the codebase to use as the canonical name of the module our own fork: https://github.com/ooni/go-libtor.

Next steps for this is that I need to test that everything works as expected and there aren't any further surprises, but I believe this task is converging.
